src/components/FilterPanel/FilterPanelCheckboxWithOverflowMenu/FilterPanelCheckboxWithOverflowMenu.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * @file Filter panel checkbox with overflow menu component.
+ * @copyright IBM Security 2020
+ */
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import classnames from 'classnames';
+
+import { getComponentNamespace } from '../../../globals/namespace';
+
+import FilterPanelCheckbox from '../FilterPanelCheckbox';
+import OverflowMenu from '../../OverflowMenu';
+
+export const namespace = getComponentNamespace(
+  'filter-panel-checkbox-with-overflow-menu'
+);
+
+const FilterPanelCheckboxWithOverflowMenu = ({
+  className,
+  checkboxClassName,
+  checkboxWrapperClassName,
+  overflowMenuAriaLabel,
+  overflowMenuClassName,
+  overflowMenuOptionsClassName,
+  children,
+  ...other
+}) => {
+  const [showOverflowButton, setShowOverflowButton] = React.useState(false);
+  const [onBlurTimeout, setOnBlurTimeout] = React.useState(undefined);
+
+  // Clear any timeouts before unmounting.
+  React.useEffect(() => () => clearTimeout(onBlurTimeout), [onBlurTimeout]);
```

Comment:
To do a cleanup with hooks, simulating unmounting, your `useEffect` would return your cleanup function. 
[React docs](https://reactjs.org/docs/hooks-effect.html#effects-with-cleanup) for effects with cleanup.